### PR TITLE
#3416 new vaccine page to be real #2

### LINKED ISSUE #3416: Replace old VaccinePage with NewVaccinePage
## Is your feature request related to a problem? Please describe.

Make the component `NewVaccinePage` the real `VaccinePage`

## Describe the solution you'd like

As above

## Implementation

Copy and paste

## Describe alternatives you've considered

Renaming files. Not as good.

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #3416 

## Change summary

- `NewVaccineModulePage` copy pasted into `VaccinePage`
- Added `PropType` fixes
- Added use of `AfterInteractions` to make navigation more fluid.
- Removed all the buttons in `SettingsPage` since we can navigate from `VaccinePage` now.

## Testing

- [ ] Can navigate to `VaccinePage`

### Related areas to think about

N/A

### CODE REVIEW

File: src/pages/VaccinePage.js [hunk 1]
```diff
@@ -1,125 +1,196 @@
...
-
 import React from 'react';
-import { connect } from 'react-redux';
-import { StyleSheet, Text, FlatList } from 'react-native';
+import { FlatList, StyleSheet } from 'react-native';
+import moment from 'moment';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import * as Animatable from 'react-native-animatable';
+import {
```
Comment (joshxg) [inline src/pages/VaccinePage.js:8]: This is just a copy paste of `NewVaccineModulePage` and then I deleted `VaccinePage` - there are a couple things to fix in this component where we have made issues, but I have no fixed anything in here. OK maybe a prop type or two, I think

File: src/pages/SettingsPage.js [hunk 2]
```diff
@@ -46,15 +45,7 @@ const exportData = async () => {
...
-const Settings = ({
-  toRealmExplorer,
-  currentUserPasswordHash,
-  requestStorageWritePermission,
-  toFridgeDetail,
-  toNewVaccineModulePage,
-  toNewSensorPage,
-  toEditSensorPage,
-}) => {
+const Settings = ({ toRealmExplorer, currentUserPasswordHash, requestStorageWritePermission }) => {
```
Comment (joshxg) [inline src/pages/SettingsPage.js:48]: Removed the settings buttons to navigate around as they're not needed anymore

File: src/pages/VaccinePage.js [hunk 3]
```diff
@@ -1,125 +1,196 @@
...
+});
+
+const stateToProps = state => {
+  const { fridge } = state;
+  const { fridges } = fridge;
+  return { fridges };
+};
+
+const dispatchToProps = dispatch => ({
+  blinkSensor: macAddress => dispatch(VaccineActions.startSensorBlink(macAddress)),
```
Comment (katherine-sussol) [inline src/pages/VaccinePage.js:190]: Oh. Think I missed updating this with https://github.com/openmsupply/mobile/pull/3411 to `BlinkActions.startSensorBlink`. I suppose can be slotted in with #3428 though.

File: src/pages/VaccinePage.js [hunk 4]
```diff
@@ -1,125 +1,196 @@
...
-  maxLine: PropTypes.array.isRequired,
-  breaches: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
-  maxDomain: PropTypes.number.isRequired,
-  minDomain: PropTypes.number.isRequired,
-  fridges: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired,
-  onSelectFridge: PropTypes.func.isRequired,
-  onOpenBreachModal: PropTypes.func.isRequired,
-  onViewAdminPage: PropTypes.func.isRequired,
-  isAdmin: PropTypes.bool.isRequired,
+  fridges: PropTypes.object,
```
Comment (katherine-sussol) [inline src/pages/VaccinePage.js:160]: Hm when the page is first loaded for me (before setting up anything), I'm getting a complaint fridges is being supplied as an array to the VaccinePage. I think `FridgeReducer` has an array default for the initial state that should maybe be converted to an object? (`locationsWithASensor`)

**Review by katherine-sussol — CHANGES_REQUESTED** [review]
👍  nice!

Might also be a good time in `NewSensorStepThree` to swap over 
`dispatch(goToSettings())` to `dispatch(goToVaccines())` now (line 111)?

That and one other prop concern (my suggestion might be off but the warning is there), otherwise looks good!

File: src/pages/VaccinePage.js (hunk 4 above)
Comment (katherine-sussol) [reply src/pages/VaccinePage.js:160]: Or maybe it is supposed to allow arrays. The name fridges does suggest an array but now I think I confused myself.

Comment (joshxg) [reply src/pages/VaccinePage.js:160]: It is being passed a realm collection which has the shape:

```
{ "0": someFridge, "1": someOtherFridge } 
```

which acts as an array but is technically an object!

Comment (katherine-sussol) [reply src/pages/VaccinePage.js:160]: Right, thanks that makes a lot more sense hah. Must just be the initialState combined with my setup causing the warning then!

File: src/pages/VaccinePage.js (hunk 3 above)
Comment (joshxg) [reply src/pages/VaccinePage.js]: b466ddc

File: src/pages/VaccinePage.js (hunk 4 above)
Comment (joshxg) [reply src/pages/VaccinePage.js]: d18f503

**Comment by joshxg** [comment]
Regarding the navigation action: it's done in I think 2 PR's down the line heh

**Review by katherine-sussol — APPROVED** [review]
🎉

**Pull request merged**